coml/core.py

```diff
@@ -229,3 +231,14 @@ def output_sanity_check(
         if "CORRECT" in last_line.upper():
             return True, reason
         raise ValueError("Unable to parse the response.")
+    
+    def vis_check(self, request:str, previous_code:str, svg_string: str, variable_descriptions: dict[str, str],source):
+        verifications = self.vis_verifier.verify(request, previous_code, svg_string, variable_descriptions,source)
+        pass_verify = all([verification['answer'] for verification in verifications])
+        reason = ""
+        for verification in verifications:
+            answer = "✅" if verification["answer"] else "❌"
```

Comment:
We don't add emojis in `core.py`. Those belong to frontend.

What we have to do here is to create a clear interface for `vis_check`. For example, its output signature can be `tuple[bool, str]`, or `tuple[Literal[...], str]`. Then in `magics.py` they can be mapped to emojis.

Reply:
`List[tuple[bool, str]]` is used to represent reason now.